core/src/main/java/org/kohsuke/stapler/framework/io/LargeText.java

```diff
@@ -303,21 +306,63 @@ private void doProgressTextImpl(StaplerRequest2 req, StaplerResponse2 rsp) throw
             start = Long.parseLong(s);
         }
 
-        if (source.length() < start) {
+        long length = source.length();
+
+        if (start > length) {
             start = 0; // text rolled over
         }
 
-        CharSpool spool = new CharSpool();
-        long r = writeLogTo(start, spool);
-
-        rsp.addHeader("X-Text-Size", String.valueOf(r));
+        rsp.addHeader("X-Text-Size", String.valueOf(length));
```

Comment:
Note that this value refers to the end offset we write to (e.g., file size), _not_ the content length which is `length - start`.